pkg/pillar/devicenetwork/dns.go

```diff
@@ -150,21 +154,24 @@ func ResolveWithPortsLambda(domain string,
 						errs = append(errs, err)
 					}
 					if response != nil {
-						resolvedIPsChan <- response
+						select {
+						case <-quit:
+						case resolvedIPsChan <- response:
+						}
 					}
-					<-work
-					wg.Done()
 				}(dnsIPCopy, srcIPCopy)
 			}
 		}
 	}
 
 	wgChan := make(chan struct{})
+
 	go func() {
 		wg.Wait()
 		close(wgChan)
 	}()
 
+	defer close(quit)
```

Comment:
Did you see this leak in the `sigusr1` log file we investigated earlier?

Reply:
I see it now.
There are goroutines that are stuck in line 164 waiting for `wg.Wait()` and because `quit` was not closed in all cases, not all `wg.Done()` were called.

Reply:
Ok, I see. 19170 occurrences.